tests/test_reader.py

```diff
@@ -52,6 +52,49 @@ def test_dask_closed(single_nd2):
     assert isinstance(dsk.compute(), np.ndarray)
 
 
+@pytest.fixture(
+    params=[
+        (None, ((2,), (32,), (32,))),
+        (2, ((2,), (2,) * 16, (2,) * 16)),
+        (3, ((2,), (3,) * 10 + (2,), (3,) * 10 + (2,))),
+        ((3, 17, 33), ((2,), (17, 15), (32,))),
+        ((2, 16, 16), ((2,), (16, 16), (16, 16))),
+        (((1, 1), (8, 8, 8, 8), (16, 16)), ((1, 1), (8, 8, 8, 8), (16, 16))),
+        (((2,), (20, 12), (32,)), ((2,), (20, 12), (32,))),
+    ],
+    ids=lambda x: str(x),
+)
+def passing_frame_chunks(request):
+    return request.param
+
+
+def test_dask_chunking(single_nd2, passing_frame_chunks):
```

Comment:
rather than creating a fixture that simply passes through the parameters, let's just directly parametrize the corresponding test (it's a little clearer that way that the test itself is parametrized):

```suggestion
FRAME_CHUNKS = [
    (None, ((2,), (32,), (32,))),
    (2, ((2,), (2,) * 16, (2,) * 16)),
    (3, ((2,), (3,) * 10 + (2,), (3,) * 10 + (2,))),
    ((3, 17, 33), ((2,), (17, 15), (32,))),
    ((2, 16, 16), ((2,), (16, 16), (16, 16))),
    (((1, 1), (8, 8, 8, 8), (16, 16)), ((1, 1), (8, 8, 8, 8), (16, 16))),
    (((2,), (20, 12), (32,)), ((2,), (20, 12), (32,))),
]


@pytest.mark.parametrize("frame_chunks, expected_chunks", FRAME_CHUNKS)
def test_dask_chunking(single_nd2, frame_chunks, expected_chunks):
```

(it also seems like tests are currently failing with `AttributeError: 'SubRequest' object has no attribute 'params'`

Reply:
I knew `@pytest.mark.parametrize` exists, but I did not find it :D changed it now